fccodegen/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""
+Main module for Web Lab code generation
+"""
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+
+#
+# Version info: Remember to keep this in sync with setup.py!
+#
+import sys
+VERSION_INT = 0, 0, 1
+VERSION = '.'.join([str(x) for x in VERSION_INT])
+if sys.version_info[0] < 3:     # pragma: no python 3 cover
+    del(x)  # Before Python3, list comprehension iterators leaked
+del(sys)
+
+
+#
+# Expose version number
+#
+def version(formatted=False):
+    """
+    Returns the version number, as a 3-part integer (major, minor, revision).
+    If ``formatedd=True``, it returns a string formatted version (e.g.
+    "codegen 1.0.0").
+    """
+    if formatted:
+        return 'fccodegen ' + VERSION
+    else:
+        return VERSION_INT
+
+
+#
+# Data directory
+#
+import os, inspect  # noqa
+try:
+    frame = inspect.currentframe()
+    DIR_ROOT = os.path.dirname(inspect.getfile(frame))
+finally:
+    # Always manually delete frame
+    # https://docs.python.org/2/library/inspect.html#the-interpreter-stack
+    del(frame)
+
+# Template directory
+DIR_TEMPLATE = os.path.join(DIR_ROOT, 'templates')
+del(os, inspect)    # Don't expose as part of codegen
+
+
+#
+# Load main classes and functions
+#
+from ._generator import (   # noqa
```

Comment:
Why the underscore on the module name? I tend to leave things public unless there's a reason not to.

Reply:
Because I want its contents to be accessible as `fccodegen.x()`, so knowing about the existence of the `generator` module has no value to the user and it can be hidden

Reply:
Unless another package might benefit from being able to use some of your internal machinery, rather than the main public methods...

It's a toss-up: restricting your public API means you're more free to change the internals. But provided you document what's stable and what's not, then caveat emptor.

Reply:
Is it ok if I keep this in mind and pick it up later once the project has more structure?

Reply:
Yep